### DISCUSSION TITLE
Best way to search only in a specific field of a tab separated file

### DISCUSSION THREAD

**Discussion by datatraveller1** [BurntSushi/ripgrep#1959] [General] [answered]
I'm looking for the best way to search only in **field no. 4** of a tab separated text file (no tabs or newlines inside a field).

The best way I have found is:
`rg -i "^([^\t]*\t){3}[^\t]*search_string" infile_tab.txt`

The regular expression looks a bit weird but seems to work properly. Field no. 4 is the fourth field (counting from 1) so {3} has to be used (number of the wished field minus one). "search_string" is searched at any position in field no. 4.

I could use `xsv search -i -s <field_name>` ... as well but xsv is optimized for comma separated files and much slower for big files than ripgrep. Any comments?

**Comment by BurntSushi** [comment, marked as answer]
You could also use `cut`. Otherwise, looks pretty good to me.

**Reply by datatraveller1** [reply]
Thank you very much. I don't want to cut fields but search in a field and print the whole line.
I think` ripgrep` is better than `cut` :-)
